### PR TITLE
always create user in the SAML back-end and update the attributes

### PR DESCRIPTION
**Pull request by schiessle**
create user in the SAML back-end and update the attributes when the user was found on another back-end during login

fix https://github.com/nextcloud/user_saml/issues/268

@maghog @devi69 could you try to apply this two line fix to your installation and report back if this solves the problem? This might also fix https://github.com/nextcloud/user_saml/issues/282

### CODE REVIEW

**Comment by maghog** [comment]
This fix did not solve the problem...

File: lib/Controller/SAMLController.php [hunk 1]
```diff
@@ -144,6 +144,8 @@ private function autoprovisionIfPossible(array $auth) {
 				// help with it and make the user known
 				$this->userManager->search($uid);
 				if($this->userManager->userExists($uid)) {
+					$this->userBackend->createUserIfNotExists($uid, $auth);
```
Comment (blizzz) [inline lib/Controller/SAMLController.php]: my gut feeling is not positive about that line

**Comment by schiessle** [comment]
@blizzz I updated the PR to only initialize the home directory of the user. Because you are right, creating the user in the saml back-end doesn't make sense if the user already exists in the ldap back-end.

What do you think?

**Comment by blizzz** [comment]
@schiessle Uff… why does it not suffice having it with the user session? I am not a fan of code duplication, especially across repos… this is not really maintainable. If it is necessary, perhaps we can move it into a trait or public method on a service. But rather I prefer to keep user backends out of dealing with storage aspects. Do you know why the user session logic does not kick in?

**Comment by schiessle** [comment]
@blizzz because in the session it happens in "prepareUserLogin()" which never happens if login happens on a external service.

**Comment by blizzz** [comment]
@schiessle  but, does it circumvent the whole LoginController::tryLogin()?

**Comment by jsalatiel** [comment]
Is https://github.com/nextcloud/user_saml/issues/268 fixed by this ?

**Pull request merged**